lib/event.c

```diff
@@ -857,19 +1155,35 @@ static int fd_poll(struct event_loop *m, const struct timeval *timer_wait,
 	rcu_read_unlock();
 	rcu_assert_read_unlocked();
 
+#if !(defined(USE_EPOLL)) ||                                                   \
+	!(defined(HAVE_EPOLL_WAIT) || defined(HAVE_EPOLL_PWAIT) ||             \
+	  defined(HAVE_EPOLL_PWAIT2))
 	/* add poll pipe poker */
 	assert(count + 1 < m->handler.pfdsize);
 	m->handler.copy[count].fd = m->io_pipe[0];
 	m->handler.copy[count].events = POLLIN;
 	m->handler.copy[count].revents = 0x00;
+#endif
 
+#if defined(USE_EPOLL) &&                                                      \
+	(defined(HAVE_EPOLL_WAIT) || defined(HAVE_EPOLL_PWAIT) ||              \
+	 defined(HAVE_EPOLL_PWAIT2))
+	/* We need to deal with a signal-handling race here: we
+	 * don't want to miss a crucial signal, such as SIGTERM or SIGINT,
+	 * that may arrive just before we enter epoll_wait(). We will block the
+	 * key signals, then check whether any have arrived - if so, we return
+	 * before calling epoll_wait(). If not, we'll re-enable the signals
+	 * in the epoll_pwait() call.
+	 */
+#else
```

Comment:
I am confused what the point of this #ifdef is. Was there code here that got removed at some point? In any case, duplicating this comment and changing the function names doesn't make a lot of sense. Please rewrite the comment below to cover both cases, and just have one copy of it. Also, it probably makes sense to remove the #ifdef, although I'd like to know why it's there to begin with.

Reply:
Indeed this duplicated doesn't make sense... Just wanna remind users about we are using epoll APIs rather than poll(). I have fixed it with a single comments covering both cases. Sry about confusing.